notebook/Solution/03 - Time Series Forecasting Using MLP.ipynb

```diff
@@ -0,0 +1,1693 @@
+   "outputs": [],
+   "source": [
+    "class ShampooDataset(Dataset):\n",
+    "    \"\"\"\n",
+    "    Convert input data into torch FloatTensor. \n",
+    "    Inherit Dataset class. Return length of instance when len method is called and return specific sample\n",
+    "    of data when indexed.\n",
+    "    \"\"\" \n",
+    "    def __init__(self, x, y):\n",
+    "        self.x = torch.FloatTensor(x)\n",
+    "        self.y = torch.FloatTensor(y)\n",
+    "        \n",
+    "    def __len__(self):\n",
+    "        return self.x.size(0)\n",
+    "    \n",
+    "    def __getitem__(self, idx):\n",
+    "        return self.x[idx], self.y[idx]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will write the codes to convert data into `Dataset` object and then into `DataLoader` object. We will be using a batch size of 4 for this `DataLoader` object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# declare batch size\n",
+    "batch_size = 4\n",
+    "\n",
+    "# convert into Dataset using customized class\n",
+    "train_Dataset = ShampooDataset(X_sequence_train, y_sequence_train)\n",
+    "test_Dataset = ShampooDataset(X_sequence_test, y_sequence_test)\n",
+    "\n",
+    "# convert into DataLoader\n",
+    "train_loader = DataLoader(train_Dataset, batch_size=batch_size, shuffle=True)\n",
+    "test_loader = DataLoader(test_Dataset, batch_size=batch_size, shuffle=False)\n",
+    "\n",
+    "# declare a dict object\n",
+    "dataloaders = {'train': train_loader, 'test': test_loader} "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It's about time to declare our MLP model architecture. The following code demonstrates how to build a simple MLP model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MLP(nn.Module):\n",
+    "    \"\"\"MLP model with configurable input size and output size.\"\"\"\n",
+    "    def __init__(self, input_size, output_size):\n",
+    "        super(MLP, self).__init__()\n",
+    "        self.input = nn.Linear(input_size, 64)\n",
+    "        self.linear1 = nn.Linear(64, 16)\n",
+    "        self.output = nn.Linear(16, output_size)\n",
+    "    \n",
+    "    def forward(self, x):\n",
+    "        x = x.view(-1, 1, 3)\n",
+    "        out = F.relu(self.input(x))\n",
+    "        out = F.relu(self.linear1(out))\n",
+    "        out = self.output(out)\n",
+    "        return out"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before we can start our training, we also need to configure some hyperparameters. Let's do that."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# setting hyperparameter\n",
+    "input_size = 3\n",
```

Comment:
input size  =  3 is because the window sequence = 3 ? might need to add a comment on this

Reply:
OK